src/iamserver/nodeinfoprovider.cpp

```diff
@@ -0,0 +1,313 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <filesystem>
+#include <fstream>
+#include <sys/statfs.h>
+
+#include <Poco/StringTokenizer.h>
+
+#include "log.hpp"
+#include "nodeinfoprovider.hpp"
+
+/***********************************************************************************************************************
+ * Constants
+ **********************************************************************************************************************/
+
+constexpr auto cProvisioningStatusPath = "/var/aos/.provisionstate";
+constexpr auto cCPUInfoPath            = "/proc/cpuinfo";
+constexpr auto cMemInfoPath            = "/proc/meminfo";
+const uint64_t cBytesPerKB             = 1024;
+
+/***********************************************************************************************************************
+ * Static
+ **********************************************************************************************************************/
+
+namespace {
+struct KeyValue {
+    std::string mKey;
+    std::string mValue;
+};
+
+std::optional<KeyValue> ParseKeyValue(const std::string& line, const std::string& delimiter = ":")
+{
+    Poco::StringTokenizer tokenizer(
+        line, delimiter, Poco::StringTokenizer::TOK_TRIM | Poco::StringTokenizer::TOK_IGNORE_EMPTY);
+
+    if (tokenizer.count() != 2) {
+        return std::nullopt;
+    }
+
+    return KeyValue {tokenizer[0], tokenizer[1]};
+}
+
+class CPUInfoParser {
```

Comment:
in my opinion, this should not be a class, so I think we move this functionality to utility and move it to a separate namespace. To get CPU info I need to create an object and call the method

Reply:
I moved it to a separate utils
Though, I kept this class in an anonymous namespace